core/src/cluster_slots_service/cluster_slots.rs

```diff
@@ -200,17 +200,21 @@ impl ClusterSlots {
         &self,
         slot: Slot,
         repair_peers: &[ContactInfo],
-    ) -> Vec<(u64, usize)> {
-        self.lookup(slot)
-            .map(|slot_peers| {
-                let slot_peers = slot_peers.read().unwrap();
-                repair_peers
-                    .iter()
-                    .enumerate()
-                    .filter_map(|(i, ci)| Some((slot_peers.get(ci.pubkey())? + 1, i)))
-                    .collect()
-            })
-            .unwrap_or_default()
+    ) -> (Vec<u64>, Vec<usize>) {
+        let mut weights = Vec::with_capacity(repair_peers.len());
+        let mut indices = Vec::with_capacity(repair_peers.len());
+
+        let Some(slot_peers) = self.lookup(slot) else {
+            return (weights, indices);
+        };
+        let slot_peers = slot_peers.read().unwrap();
+        for (index, peer) in repair_peers.iter().enumerate() {
+            if let Some(stake) = slot_peers.get(peer.pubkey()) {
+                weights.push(stake + 1);
```

Comment:
am i right that this +1 is to avoid 0 weight situation? Can it even happen if Stake is Some?

Reply:
the +1 is part of logic, I'm not touching logic in this PR. Whether it is a good idea I doubt it does anything useful whatsoever.